benchmark/rgw_benchmark_run.yml

```diff
@@ -0,0 +1,16 @@
+---
+- hosts: benchmark_hosts
+  become: True
+  gather_facts: True
+  tasks:
+    - name: Run hummingbird bench  and record version/config info
+      shell: 'echo -e "\nHumminbird Version: `/opt/ceph_bench/hummingbird version`\n\nBench Config Used:\n`cat /opt/ceph_bench/{{ item.name }}`\n\nHummingbird Benchmark Output:\n`/opt/ceph_bench/hummingbird bench /opt/ceph_bench/{{ item.name }}`" | tee /opt/ceph_bench/logs/{{ item.name }}.{{ ansible_date_time.iso8601_basic_short }}.log'
```

Comment:
Can we convert this from shell to a template?

Reply:
I'm fairly new to ansible and I've seen examples of templates used to write out config files but I don't understand what you mean in this context.   Can you give me more details

Reply:
Here is what I was thinking.  You could do something like this

```
  vars:
    tests:
      - name: foo
      - name: bar
      - name: baz
  tasks:
    - name: Check for logfile
      stat:
        path: "/tmp/{{ item.name }}.{{ ansible_date_time.iso8601_basic_short }}.log"
      register: logfile
      with_items: "{{ tests }}"

    - name: fail when logfile exists
      fail:
        msg: "Log file already exists"
      when: item.stat.exists
      with_items: "{{ logfile.results }}"

    - name: Write logfile
      template:
        src: ceph_bench.j2
        dest: "/tmp/{{ item.name }}.{{ ansible_date_time.iso8601_basic_short }}.log"
      with_items: "{{ tests }}"
      register: mytemplate

    - name: Get test output
      command: "cat /tmp/{{ item.name }}.{{ ansible_date_time.iso8601_basic_short }}.log"
      register: benchout
      with_items: "{{ tests }}"

    - debug:
        var: item.stdout_lines
      with_items: "{{ benchout.results }}"
```

and the template looks like this:

```
Humminbird Version: {{ lookup('pipe', 'ansible --version') }}

Bench Config Used:
{{ lookup('file', './junk.txt') }}

Hummingbird Benchmark Output:
{{ lookup('pipe', 'echo /opt/ceph_bench/' + item.name) }}
```

I used `ansible --version` and `echo` as the pipe commands running locally.

Reply:
You could also just template the header of the file and leave the last lookup to run the benchmark out of it and still have the shell module run `hummingbird bench /opt/ceph_bench/{{ item.name }} >> ...log` to make it clear where/when the bench mark commands are running.

Reply:
Wow!   thanks for the example.    It seems a simpler to just you the shell command IMHO.

Reply:
We can probably simplify that, there is only 1 file in with_items, so we don't need to use with_items at all.
Just change (when: item.run_type to when: rgw_bench)

  - name: Run hummingbird bench and record output
     shell: '/opt/ceph_bench/hummingbird bench /opt/ceph_bench/rgw_bench_test.conf'
     register: hummingbird_bench_output

- name: Template log file
   template:
     src: ceph_bench.j2
     dest: "/opt/ceph_bench/logs/rgw_bench_test.{{ ansible_date_time.iso8601_basic_short }}.log"
     backup: true

Then if the template looks similar to what Mike had above, except change the first lookup to look for /opt/ceph_bench/rgw_bench.conf and the last lookup to just be {{ hummingbird_bench_output.stdout }}
That should work fine.

We can probably skip the check for logfile etc - whilst I can never see that being used with ansible_date, we can add "backup: true" to the template and it'd ensure it doesn't overwrite log files.